packages/database/src/SqlQuery.js

```diff
@@ -11,9 +11,18 @@ export class SqlQuery {
   /**
    * @public
    * @param {unknown[]} arr
+   * @param {string} type type of array (if the array may be empty postgres requires typecasting)
    * @returns {string} SQL parameter injection string for the array of values
    */
-  static array = arr => `(${arr.map(() => '?').join(',')})`;
+  static array = (arr, type) =>
+    `ARRAY[${arr.map(() => '?').join(',')}]${type ? `::${type}[]` : ''}`;
+
+  /**
+   * @public
+   * @param {unknown[]} arr
+   * @returns {string} SQL parameter injection string for the array of values
+   */
+  static record = arr => `(${arr.map(() => '?').join(',')})`;
```

Comment:
Bit of a rename here, this whole time we've actually been creating 'records' rather than arrays! https://www.postgresqltutorial.com/postgresql-plpgsql/plpgsql-record-types/